lib/bolt/executor.rb

```diff
@@ -184,5 +190,24 @@ def file_upload(targets, source, destination, options = {}, &callback)
         results
       end
     end
+
+    # Plan context doesn't make sense for most transports but it is tightly
+    # coupled with the orchestrator transport since the transport behaves
+    # differently when a plan is running. In order to limit how much this
+    # pollutes the transport API we only handle the orchestrator transport here.
+    # Since we callt this function without resolving targets this will result
+    # in the orchestrator transport always being initialized during plan runs.
+    # For now that's ok.
+    #
+    # In the future if other transports need this or if we want a plan stack
+    # we'll need to refactor.
+    def start_plan(plan_context)
+      transport('pcp').plan_context = plan_context
+      @plan_logging = true
```

Comment:
Should this get set back to false in `finish_plan` so it's not still true for any other tasks that get run? Or does the executor get instantiated anew for each job run? I can't think of an exception where a plan then a task would run separately as part of one execution, but that might exist.

Reply:
finish_plan is called by the CLI so no more tasks will run.